starlark/eval.go

```diff
@@ -836,14 +826,50 @@ unknown:
 	return nil, fmt.Errorf("unknown binary op: %s %s %s", x.Type(), op, y.Type())
 }
 
-func repeat(elems []Value, n int) (res []Value) {
-	if n > 0 {
-		res = make([]Value, 0, len(elems)*n)
-		for i := 0; i < n; i++ {
-			res = append(res, elems...)
-		}
+// It's always possible to overeat in small bites but we'll
+// try to stop someone swallowing the world in one gulp.
+const maxAlloc = 10000000
+
+func tupleRepeat(elems Tuple, n Int) (Tuple, error) {
+	if len(elems) == 0 {
+		return nil, nil
+	}
+	i, err := AsInt32(n)
+	if err != nil {
+		return nil, fmt.Errorf("repeat count %s too large", n)
+	}
+	if i < 1 {
+		return nil, err
```

Comment:
`nil, nil` would be more clear.

Reply:
It sure would. Thanks, that was an oversight.